packages/mask/src/plugins/Savings/protocols/LDOProtocol.ts

```diff
@@ -0,0 +1,185 @@
+    public decimals = 18
+    public apr = '0.00'
+    public balance = new BigNumber('0')
+    public availableNetworks: SavingsNetwork[] = [
+        {
+            chainId: ChainId.Mainnet,
+            chainName: 'Ethereum',
+            contractAddress: getSavingsConstants(ChainId.Mainnet).LIDO_STETH || '',
+        },
+        {
+            chainId: ChainId.Gorli,
+            chainName: 'Gorli',
+            contractAddress: getSavingsConstants(ChainId.Gorli).LIDO_STETH || '',
+        },
+    ]
+
+    public getFungibleTokenDetails(chainId: ChainId): FungibleTokenDetailed {
+        let contractAddress = ''
+
+        for (const network of this.availableNetworks) {
+            if (network.chainId === chainId) {
+                contractAddress = network.contractAddress
+            }
+        }
+
+        return {
+            type: 1,
+            chainId: chainId,
+            address: contractAddress,
+            symbol: 'stETH',
+            decimals: 18,
+            name: 'Liquid staked Ether 2.0',
+            logoURI: [
+                'https://static.debank.com/image/eth_token/logo_url/0xae7ab96520de3a18e5e111b5eaab095312d7fe84/f768023f77be7a2ea23c37f25b272048.png',
+                'https://tokens.1inch.io/0xae7ab96520de3a18e5e111b5eaab095312d7fe84.png',
+            ],
+        }
+    }
+
+    public async getApr() {
+        try {
+            const LidoAprUrl = 'https://cors.r2d2.to/?https://stake.lido.fi/api/steth-apr'
+            const response = await fetch(LidoAprUrl)
+            const apr = await response.text()
+            this.apr = apr
+            return apr
+        } catch (error) {
+            console.log('LDO `getApr()` error', error)
+            // Default APR is 5.30%
+            this.apr = '5.30'
+            return '5.30'
+        }
+    }
+
+    public async getBalance(chainId: ChainId, web3: Web3, account: string) {
+        try {
+            const contract = createContract<Lido>(
+                web3,
+                getSavingsConstants(chainId).LIDO_STETH || '',
+                LidoABI as AbiItem[],
+            )
+            const balance = await contract?.methods.balanceOf(account).call()
+            this.balance = new BigNumber(balance || '0')
+            return this.balance
+        } catch (error) {
+            console.log('LDO `getBalance()` error', error)
+            this.balance = new BigNumber('0')
+            return this.balance
+        }
+    }
+
+    public async depositEstimate(account: string, chainId: ChainId, web3: Web3, value: BigNumber.Value) {
+        try {
+            const contract = createContract<Lido>(
+                web3,
+                getSavingsConstants(chainId).LIDO_STETH || '',
+                LidoABI as AbiItem[],
+            )
+            const gasEstimate = await contract?.methods
+                .submit(
+                    getSavingsConstants(chainId).LIDO_REFERRAL_ADDRESS || '0x0000000000000000000000000000000000000000',
```

Comment:
```suggestion
                    getSavingsConstants(chainId).LIDO_REFERRAL_ADDRESS || ZERO_ADDRESS
```

We have a constant `ZERO_ADDRESS` which you can import from `@masknet/web3-shared-evm` package. Please replace other literal zero addresses with it as well.